### PR TITLE
Preliminary cache migration code

### PR DESCRIPTION
**Pull request by aphelionz**
This PR adds `OrbitDb.attemptMigration`

This handles the automatic migration the cache from multiple LevelDB stores (pre-0.21) to a single LevelDB store with multiple keys. (0.22+)

- [X] Tests for "default cache" scenario
- [x] Tests for passing the `{ directory }` option into `OrbitDB.create`

### CODE REVIEW

File: src/OrbitDB.js [hunk 1]
```diff
@@ -385,6 +387,40 @@ class OrbitDB {
     logger.debug(`Saved manifest to IPFS as '${dbAddress.root}'`)
   }
 
+  async _attemptMigration(options, dbAddress) {
+    let oldCache = this.caches[options.directory], oldStore
```
Comment (haadcode) [inline src/OrbitDB.js]: I'm confused of the `,` after the assignment. How about declaring `oldStore` explicitly and putting on its own line?

**Comment by haadcode** [comment]
Looks good to me, generally. One little stylistic comment. I'm still wondering though, if it would be possible to move the cache migration to its own file, and isolate it completely out of the main class?

**Comment by aphelionz** [comment]
@haadcode Maybe we should start implementing a proper migrations folder. We could start with @shamb0t's 0.19-0.20, 0.20-0.21 migrations and include this as 0.21-0.22

**Comment by aphelionz** [comment]
And then simply have some call to this.migrate() at places in the OrbitDB class

**Comment by haadcode** [comment]
@aphelionz that'd make sense 👍

**Comment by aphelionz** [comment]
@haadcode @shamb0t How's that look?

**Comment by haadcode** [comment]
> @haadcode @shamb0t How's that look?

LGTM 👍

**Comment by shamb0t** [comment]
LGTM too :+1:

File: src/migrations/0.21-0.22.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+const fs = require('fs')
+
+const Cache = require('orbit-db-cache')
+
+const Logger = require('logplease')
+const logger = Logger.create('orbit-db')
+Logger.setLogLevel('ERROR')
+
+async function migrate (OrbitDB, options, dbAddress) {
+  let oldCache = OrbitDB.caches[options.directory]; let oldStore
```
Comment (shamb0t) [inline src/migrations/0.21-0.22.js]: Any reason not to put `let oldStore` on its own line?

Comment (aphelionz) [reply src/migrations/0.21-0.22.js]: ha, actually `standard --fix` changed that from a comma to a semicolon and left it on the same line

Comment (aphelionz) [reply src/migrations/0.21-0.22.js]: I'll move it down

Comment (shamb0t) [reply src/migrations/0.21-0.22.js]: Oh sneaky `standard`

**Pull request merged**